src/utils/date.ts

```diff
@@ -70,6 +73,33 @@ export function dateFromISO(iso8601: string | Date, isEndDate = false): Date | n
   return date;
 }
 
+/**
+ * Parse a localized date string into a valid Date.
+ * return false if date is invalid, or out of range
+ *
+ * @param value
+ * @param localeData
+ */
+export function dateFromLocalizedString(value: string, localeData: DateLocaleData): Date | false {
+  if (!localeData) {
+    return false;
```

Comment:
Returning a boolean from a function that indicates a date is returned seems odd to me. Let's return `null` or `undefined` instead.